### PR TITLE
Add patch method to user_profile_controller.rb.

### PR DESCRIPTION
**Pull request by sawa-zen**
@rrreeeyyy 

Please review the patch method. 🙇

### CODE REVIEW

File: spec/controllers/api/v1/user_profile_controller_spec.rb [hunk 1]
```diff
@@ -43,4 +43,22 @@
...
+
+  describe "PATCH /" do
+    let(:method) { patch }
+    let(:path) { "/" }
+    let!(:params) { FactoryGirl.build(:user_profile).slice(:screen_name, :description).merge({ content: "data:text/plain;base64,base64encodedstring" }) }
+    it_should_behave_like "authorization!"
+
+    it "return 201 if updated" do
+      patch path, params.to_json, valid_header(user_profile.user.name)
+      expect(last_response.status).to eq 201
```
Comment (rrreeeyyy) [inline spec/controllers/api/v1/user_profile_controller_spec.rb]: I think it should return 200.

Comment (rrreeeyyy) [inline spec/controllers/api/v1/user_profile_controller_spec.rb]: and I think need more test like really parameter updated?

File: spec/controllers/api/v1/user_profile_controller_spec.rb [hunk 2]
```diff
@@ -43,4 +43,22 @@
...
+    end
+
+    it "return 201 and updated if empty parameters" do
+      patch path, {}.to_json, valid_header(user_profile.user.name)
+      expect(user_profile.reload).to eq user_profile
+      expect(last_response.status).to eq 201
```
Comment (rrreeeyyy) [inline spec/controllers/api/v1/user_profile_controller_spec.rb]: it should retun 304 or 200?

File: app/controllers/api/v1/user_profile_controller.rb [hunk 3]
```diff
@@ -34,6 +34,23 @@ def parse_data_url(data)
...
+
+        if data[:content]
+          data_uri = parse_data_url(data[:content])
+          data[:image] = StringIO.new(data_uri.data)
+          data[:image_content_type] = data_uri.content_type
+          data.delete(:content)
+        end
+
+        user_profile.update_attributes(data)
+        json = ::Api::Resources::UserProfileResource.new(user_profile).to_json
```
Comment (rrreeeyyy) [inline app/controllers/api/v1/user_profile_controller.rb]: this line cause known bugs refs #74 probably.

**Comment by sawa-zen** [comment]
@rrreeeyyy 
Fixed status and test. Please review again. 🙇‍♀️

**Comment by sawa-zen** [comment]
@rrreeeyyy  🙇

**Pull request merged**